java/opendcs/src/main/java/opendcs/dao/EnumSqlDao.java

```diff
@@ -15,64 +15,81 @@
 */
 package opendcs.dao;
 
-import java.sql.Connection;
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Iterator;
 import java.util.List;
-import java.util.Objects;
 import java.util.Optional;
 import java.util.stream.Collectors;
 
 import org.jdbi.v3.core.Handle;
-import org.opendcs.database.SimpleTransaction;
+import org.jdbi.v3.core.Jdbi;
+import org.opendcs.database.JdbiTransaction;
+import org.opendcs.database.TransactionContextImpl;
 import org.opendcs.database.api.DataTransaction;
 import org.opendcs.database.api.DatabaseEngine;
 import org.opendcs.database.api.OpenDcsDataException;
-import org.opendcs.database.api.OpenDcsDatabase;
+import org.opendcs.database.impl.opendcs.jdbi.column.databasekey.DatabaseKeyArgumentFactory;
+import org.opendcs.database.impl.opendcs.jdbi.column.databasekey.DatabaseKeyColumnMapper;
 import org.opendcs.database.model.mappers.dbenum.DbEnumBuilderMapper;
 import org.opendcs.database.model.mappers.dbenum.DbEnumBuilderReducer;
 import org.opendcs.database.model.mappers.dbenum.EnumValueMapper;
 import org.opendcs.utils.logging.OpenDcsLoggerFactory;
 import org.opendcs.utils.sql.GenericColumns;
 import org.opendcs.utils.sql.SqlKeywords;
+
+import org.openide.util.lookup.ServiceProvider;
+
 import org.slf4j.Logger;
 
 import decodes.db.EnumValue;
 import decodes.db.ValueNotFoundException;
 import decodes.db.DbEnum.DbEnumBuilder;
 import opendcs.dai.EnumDAI;
-
+import decodes.db.DatabaseException;
 import decodes.db.DbEnum;
 import decodes.db.EnumList;
 import decodes.sql.DbKey;
 import decodes.sql.DecodesDatabaseVersion;
+import decodes.sql.KeyGenerator;
 import decodes.tsdb.DbIoException;
+import decodes.util.DecodesSettings;
 
 /**
  * Data Access Object for writing/reading DbEnum objects to/from a SQL database
  * @author mmaloney Mike Maloney, Cove Software, LLC
  */
+@ServiceProvider(service = EnumDAI.class, path = "DEFAULT")
```

Comment:
"DEFAULT" seems equivalent to a null path, which implies support for XML. Should be typed to the different implementations:
```
@ServiceProviders({
		@ServiceProvider(service = EnumDAI.class, path = "OPENTSDB"),
		@ServiceProvider(service = EnumDAI.class, path = "CWMS"),
		@ServiceProvider(service = EnumDAI.class, path = "HDB"),		
})
```

For a case that we actually want a default fallback, adding the no path makes sense:
```
@ServiceProviders({
		@ServiceProvider(service = EnumDAI.class),
		@ServiceProvider(service = EnumDAI.class, path = "OPENTSDB"),
		@ServiceProvider(service = EnumDAI.class, path = "CWMS"),
		@ServiceProvider(service = EnumDAI.class, path = "HDB"),		
})
```